src/eval.rs

```diff
@@ -353,3 +345,76 @@ pub fn eval_entry<'tcx>(
         Err(e) => report_error(&ecx, e),
     }
 }
+
+/// Turns an array of arguments into a Windows command line string.
+///
+/// The string will be UTF-16 encoded and NUL terminated.
+///
+/// Panics if the zeroth argument contains the `"` character because doublequotes
+/// in argv[0] cannot be encoded using the standard command line parsing rules.
+fn args_to_utf16_command_string<I, T>(mut args: I) -> Vec<u16>
+where
+    I: Iterator<Item = T>,
+    T: AsRef<str>,
+{
+    // Parse argv[0]. Slashes aren't escaped. Literal double quotes are not allowed.
+    let mut cmd = if let Some(arg0) = args.next() {
+        let arg0 = arg0.as_ref();
+        if arg0.is_empty() {
+            "\"\"".into()
+        } else if arg0.contains('"') {
+            panic!("argv[0] cannot contain a doublequote (\") character");
+        } else {
+            // Always surround argv[0] with quotes.
+            let mut s = String::new();
+            s.push('"');
+            s.push_str(arg0);
+            s.push('"');
+            s
+        }
+    } else {
+        return vec![0];
+    };
+
+    // Build the other arguments.
+    for arg in args {
+        let arg = arg.as_ref();
+        cmd.push(' ');
+        if arg.is_empty() {
+            cmd.push_str("\"\"");
+        } else if !arg.bytes().any(|c| matches!(c, b'"' | b'\t' | b' ')) {
+            cmd.push_str(arg);
```

Comment:
```suggestion
            // No quote, tab, or space -- no escaping required.
            cmd.push_str(arg);
```

Reply:
The fact that we don't have to escape even when there are `\` is quite surprising and might be worth a comment; this is again that "Backslashes are interpreted literally, unless they immediately precede a double quote mark" clause.

Reply:
Maybe I should explain this in the next else block? Something like:

```rust
} else {
    // Spaces and tabs are escaped by surrounding them in quotes.
    // Quotes are themselves escaped by using backslashes when in a
    // quoted block.
    // Backslashes only need to be escaped when one or more are directly
    // followed by a quote. Otherwise they are taken literally.
```

Reply:
Sounds good!